src/hooks/auditoriumSections.tsx

```diff
@@ -0,0 +1,207 @@
+      },
+    ];
+  });
+};
+
+export interface UseAuditoriumSectionProps {
+  rows: number;
+  columns: number;
+  videoWidthInSeats: number;
+  videoHeightInSeats: number;
+  sectionId?: string;
+  venueId?: string;
+}
+
+export const useAuditoriumSection = ({
+  venueId,
+  sectionId,
+  rows,
+  columns,
+  videoWidthInSeats,
+  videoHeightInSeats,
+}: UseAuditoriumSectionProps) => {
+  useConnectAuditoriumSections(venueId);
+
+  const { userWithId } = useUser();
+  const userId = userWithId?.id;
+
+  const sectionsById = useSelector(currentAuditoriumSectionsByIdSelector);
+  const section = sectionId ? sectionsById?.[sectionId] : undefined;
+
+  const seatedUsers = useSectionSeatedUsers(venueId, sectionId);
+
+  const getUserBySeat = useGetUserByPosition({
+    venueId,
+    positionedUsers: seatedUsers,
+  });
+
+  const takeSeat = useCallback(
+    ({ row, column }: GridPosition) => {
+      if (!sectionId || !venueId || !userId) return;
+
+      setGridData({
+        venueId,
+        userId,
+        gridData: { sectionId, row, column },
+      });
+    },
+    [sectionId, venueId, userId]
+  );
+
+  const checkIfSeat = useCallback(
+    ({ row, column }: GridPosition) => {
+      const covertedRowCoordinate = convertCoordinate({
+        index: row,
+        totalAmount: rows,
+      });
+      const convertedColumnCoordinate = convertCoordinate({
+        index: column,
+        totalAmount: columns,
+      });
+
+      const isInVideoRow =
+        Math.abs(covertedRowCoordinate) <= videoHeightInSeats / 2;
+      const isInVideoColumn =
+        Math.abs(convertedColumnCoordinate) <= videoWidthInSeats / 2;
+
+      return !(isInVideoRow && isInVideoColumn);
+    },
+    [rows, columns, videoHeightInSeats, videoWidthInSeats]
+  );
+
+  const leaveSeat = useCallback(() => {
+    if (!venueId || !userId) return;
+
+    setGridData({ venueId, userId, gridData: undefined });
+  }, [venueId, userId]);
+
+  return {
+    auditoriumSection: section,
+    isAuditoriumSectionLoaded: isLoaded(sectionsById),
+
+    getUserBySeat,
+    takeSeat,
+    leaveSeat,
+    checkIfSeat,
+  };
+};
+
+export const useSectionSeatedUsers = (venueId?: string, sectionId?: string) => {
+  const { recentVenueUsers } = useRecentVenueUsers();
+
+  return useMemo(
+    () =>
+      recentVenueUsers.filter(
+        (user) =>
+          venueId && sectionId && user.data?.[venueId]?.sectionId === sectionId
+      ),
+    [recentVenueUsers, venueId, sectionId]
+  );
+};
+
+export const useAuditoriumSections = (venueId?: string) => {
+  useConnectAuditoriumSections(venueId);
+
+  const sections = useSelector(currentAuditoriumSectionsSelector);
+
+  return useMemo(
+    () => ({
+      auditoriumSections: sections ?? [],
+      isAuditoriumSectionsLoaded: isLoaded(sections),
+    }),
+    [sections]
+  );
+};
+
+export interface UseAuditoriumGridProps {
+  rows: number;
+  columns: number;
+  getUserBySeat: (gridData: GridPosition) => WithId<User> | undefined;
```

Comment:
If the `useGetUserBySeat` hook exported the type of it's return value, we could use it directly here.

Reply:
changed in https://github.com/sparkletown/sparkle/pull/1251/commits/16307165634c1f4b60ed8efc7a120a89c88f6a0f